pkg/async/notifications/factory.go

```diff
@@ -103,14 +103,30 @@ func NewNotificationsProcessor(config runtimeInterfaces.NotificationsConfig, sco
 			panic(err)
 		}
 		emailer = GetEmailer(config, scope)
+		return implementations.NewProcessor(sub, emailer, scope)
+	case common.GCP:
+		projectID := config.GCPConfig.ProjectID
+		subscription := config.NotificationsProcessorConfig.QueueName
+		var err error
+		err = async.Retry(reconnectAttempts, reconnectDelay, func() error {
+			sub, err = gizmoGCP.NewSubscriber(context.TODO(), projectID, subscription)
+			if err != nil {
+				logger.Warnf(context.TODO(), "Failed to initialize new gizmo gcp subscriber with config [ProjectID: %s, Subscription: %s] and err: %v", projectID, subscription, err)
+			}
+			return err
+		})
+		if err != nil {
+			panic(err)
+		}
+		emailer = GetEmailer(config, scope)
```

Comment:
out of curiosity, are you using the sendgrid emailer here? how did you set this up for testing?

Reply:
Yes, we are using SendGrid which can be purchased via GCP Marketplace (https://console.cloud.google.com/marketplace/product/sendgrid-app/sendgrid-email). After registering on SendGrid and verify sender identity we can then use that verified email.

The config is like this:
```notifications:
      emailer:
        body: |
          Execution {{ workflow.project }}/{{ workflow.domain }}/{{ workflow.name }}/{{ name }} has {{ phase }}.
          Details: https://flyte.example.com/console/projects/{{ project }}/domains/{{ domain }}/executions/{{ name }}.
          {{ error }}
        sender: <verified-email>@gojek.com
        subject: 'Flyte: {{ project }}/{{ domain }}/{{ launch_plan.name }} has {{ phase }}'
        emailServerConfig:
          serviceName: sendgrid
          apiKeyEnvVar: SENDGRID_API_KEY```
          
I then manually set SENDGRID_API_KEY as environment variable for flyteadmin container. We may also need to be able to configure flyteadmin environment variable via Helm chart as it's not supported for now: https://github.com/flyteorg/flyte/blob/master/charts/flyte/templates/admin/deployment.yaml#L91-L115

Reply:
thank you for testing! do you mind filing an issue for being able to set SENDGRID_API_KEY in the helm chart? https://github.com/flyteorg/flyte/issues/new/choose

Reply:
Here it is: https://github.com/flyteorg/flyte/issues/1520